docs/keyboard.md

```diff
@@ -224,3 +224,51 @@ static scheduleLayoutAnimation(event)
 ```
 
 Useful for syncing TextInput (or other keyboard accessory view) size of position changes with keyboard movements.
+
+## Type Definitions
+
+### KeyboardMetrics
+`KeyboardMetrics` object is part of [`Keyboard`](text#addListener) callback and contains the measurement data of `Keyboard`.
+
+
+#### Example
+
+```js
+{
+  screenX: 0,
+  screenY: 401.1428527832031,
+  width: 411.4285583496094,
+  height: 282.28570556640625,
+}
+```
+
+#### Properties
+
+| Name      | Type    | Optional  | Description                                                         |
+| --------- | ------- | --------- | ------------------------------------------------------------------- |
+| screenX   | number  | No        | Virtual keyboard X position on screen in pixels.                    |
+| screenY   | number  | No        | Virtual keyboard Y position on screen in pixels.                    |
+| width     | number  | No        | Virtual keyboard width in pixels.                                   |
+| height    | number  | No        | Virtual keyboard height in pixels.                                  |
+
+### KeyboardEvent
+
+`KeyboardEvent` object is returned in the callback as a result of a [`Keyboard`](text#addListener) event. It contains a key called `endCoordinates` with a value which is an object of type [`KeyboardMetrics`](text#KeyboardMetrics).
+
+#### Example
+
+```js
+{
+  duration: 0,
+  easing: "keyboard",
+  endCoordinates: KeyboardMetrics
+}
+```
+
+#### Properties
+
+| Name           | Type                                                                       | Optional | Description                                    |
+| -------------- | -------------------------------------------------------------------------- | -------- | ---------------------------------------------- |
+| duration       | number                                                                     | No       | Virtual keyboard animation duration.           |
+| easing         | enum(`"easeIn"`, `"easeInEaseOut"`, `"easeOut"`, `"linear"`, `"keyboard"`) | No       | Virtual keyboard animation easing function.    |
+| endCoordinates | [KeyboardMetrics](text#KeyboardMetrics) object                             | No       | Object containing `Keyboard` measurement data. |
```

Comment:
```suggestion
| duration       | number                                                                     | No       | Virtual keyboard animation duration.           |
| easing         | enum(`"easeIn"`, `"easeInEaseOut"`, `"easeOut"`, `"linear"`, `"keyboard"`) | No       | Virtual keyboard animation easing function.    |
| endCoordinates | [KeyboardMetrics](text#KeyboardMetrics) object                             | No       | Object containing `Keyboard` measurement data. |
```

Few things in this table:
* we are missing iOS only fields - ` startCoordinates` and `isEventFromThisApp`
* `easing` is always `'keyboard'` on Android
* `duration` is always `0` on Android

You can see the example of marking platform specific fields in type tables here:
* https://reactnative.dev/docs/next/image#imagesource